### PR TITLE
Bump-up the version for RoCm-6.1.2 release

### PR DESCRIPTION
**Pull request by renjithravindrankannath**

### CODE REVIEW

**Review by afzpatel — APPROVED** [review]

File: var/spack/repos/builtin/packages/rccl/package.py [hunk 1]
```diff
@@ -88,7 +90,7 @@ class Rccl(CMakePackage):
         "6.0.0",
         "6.0.2",
         "6.1.0",
-        "6.1.1",
+        "6.1.2",
     ]:
         depends_on(f"rocm-smi-lib@{ver}", when=f"@{ver}")
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rccl/package.py:78]: Can you combine the 2 for loops into 1 inside the rccl package as both version list is same

File: var/spack/repos/builtin/packages/rocm-cmake/package.py [hunk 2]
```diff
@@ -50,6 +51,7 @@ class RocmCmake(CMakePackage):
         "6.0.2",
         "6.1.0",
         "6.1.1",
+        "6.1.2",
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocm-cmake/package.py:54]: inside the rocm-cmake , can you remove the dead code -
def cache_test_sources(self):
if self.spec.satisfies("@:5.1.0"):
            return
def test(self):
        if self.spec.satisfies("@:5.1.0"):
            print("Skipping: stand-alone tests")
            return

File: var/spack/repos/builtin/packages/rocm-debug-agent/package.py [hunk 3]
```diff
@@ -71,6 +73,7 @@ class RocmDebugAgent(CMakePackage):
         "6.0.2",
         "6.1.0",
         "6.1.1",
+        "6.1.2",
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocm-debug-agent/package.py]: inside the rocm-debug-agent, can you combine the 2 for loops into one as the version list remains the same

File: var/spack/repos/builtin/packages/rocblas/package.py [hunk 4]
```diff
@@ -80,6 +81,7 @@ class Rocblas(CMakePackage):
         "6.0.2",
         "6.1.0",
         "6.1.1",
+        "6.1.2",
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocblas/package.py:83]: in the rocblas package, can you move  depends_on("rocm-cmake", type="build")
 inside the for loop.

**Review by srekolam — CHANGES_REQUESTED** [review]
LGTM overall, except few minor comments

File: var/spack/repos/builtin/packages/rccl/package.py (hunk 1 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rccl/package.py:78]: Removed redundant for loop

File: var/spack/repos/builtin/packages/rocm-debug-agent/package.py (hunk 3 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocm-debug-agent/package.py]: Removed redundant for loop

File: var/spack/repos/builtin/packages/rocblas/package.py (hunk 4 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocblas/package.py:83]: brought rocm-cmake into loop

**Review by srekolam — APPROVED** [review]

File: var/spack/repos/builtin/packages/rocm-cmake/package.py (hunk 2 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocm-cmake/package.py:54]: Removing deprecated version checks

**Comment by renjithravindrankannath** [comment]
@spackbot re-run pipeline

**Comment by renjithravindrankannath** [comment]
Hi @tldahlgren , Can you please review and merge this when you get a chance?

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256s.

**Pull request merged**